mne/epochs.py

```diff
@@ -1066,15 +1066,47 @@ def _compute_aggregate(self, picks, mode='mean'):
     @property
     def _name(self):
         """Give a nice string representation based on event ids."""
+        return self._get_name()
+
+    def _get_name(self, count='frac', ms='×', sep='+'):
+        """Generate human-readable name for epochs and evokeds from event_id.
+
+        Parameters
+        ----------
+        count : 'frac' | 'total'
+            Whether to include the fraction or total number of epochs that each
+            event type contributes to the number of all epochs.
+            Ignored if only one event type is present.
+        ms : str | None
+            The multiplication sign to use. Pass ``None`` to omit the sign.
+            Ignored if only one event type is present.
+        sep : str
+            How to separate the different events names. Ignored if only one
+            event type is present.
+        """
+        _check_option('count', value=count, allowed_values=['frac', 'total'])
+
         if len(self.event_id) == 1:
             comment = next(iter(self.event_id.keys()))
         else:
-            count = Counter(self.events[:, 2])
+            counter = Counter(self.events[:, 2])
             comments = list()
-            for key, value in self.event_id.items():
-                comments.append('%.2f × %s' % (
-                    float(count[value]) / len(self.events), key))
-            comment = ' + '.join(comments)
+
+            # Take care of padding
+            if ms is None:
+                ms = ' '
+            else:
+                ms = f' {ms} '
+
+            for event_name, event_code in self.event_id.items():
+                if count == 'frac':
+                    frac = float(counter[event_code]) / len(self.events)
+                    comment = f'{frac:.2f}{ms}{event_name}'
+                else:  # 'total'
+                    comment = f'{counter[event_code]}{ms}{event_name}'
+                comments.append(comment)
```

Comment:
this is a bit more DRY:
```suggestion
            div, fmt = 1., ''
            if count == 'frac':
                div, fmt = len(self.events), '.2f'
            for event_name, event_code in self.event_id.items():
                comment = f'{counter[event_code] / div:{fmt}}{ms}{event_name}'
                comments.append(comment)
```

Reply:
the last line does not need dedenting, it needs to be inside the loop otherwise you only get the last comment.  But it's not an important change; if you find it harder to read let's skip it.